reviewrot/basereview.py

```
@@ -82,6 +83,62 @@ def check_request_state(self, created_at,
         return True
 
+    def _decode_response(self,response):
+
+        GERRIT_JSON_PREFIX = ")]}\'\n"
+
+        """ Remove Gerrit's prefix and convert to JSON.
+        :returns:
+            Converted JSON content
+        :raises:
+            requests.HTTPError if the response contains an HTTP error status code.
+        """
+        content = response.content.strip()
+        if response.encoding:
+            content = content.decode(response.encoding)
+        response.raise_for_status()
+        content_type = response.headers.get('content-type', '')
+        if content_type.split(';')[0] != 'application/json':
+            return content
+        if content.startswith(GERRIT_JSON_PREFIX):
+            content = content[len(GERRIT_JSON_PREFIX):]
+        try:
+            return json.loads(content)
+        except ValueError:
+            logging.error('Invalid json content: %s' % content)
+            raise
+
+
+    def _call_api(self, url, method='GET', ssl_verify=True):
+        """
+        Method used to call the API.
+        It returns the raw JSON returned by the API or raises an exception
+        if something goes wrong.
+        Args:
+            method (str): the URL to call, can be GET, POST, DELETE, UPDATE...
+                          Defaults to GET
+        Returns:
+            raw JSON returned by API
+        """
+        response = self.session.request(method=method,url=url,headers=self.header,verify=ssl_verify)
+        decoded_response = None
+        try:
+
+            decoded_response = response.json()
+        except Exception as err:
+            try:
+                decoded_response = self._decode_response(response)
+            except HTTPError as http_error:
+                return ''
+            except Exception as e:
+                log.exception('Error while decoding JSON: {0}'.format(err))
+                raise
+        if response.status_code != 200:
+            if 'error_code' in decoded_response:
```

If the status_code is not 200 but there's no error_code in decoded response you silently ignore this error. So if configuration is incorrect (as I had when I was testing this PR) - there was no error message telling me that my host configuration wasn't pointing to the right place.